src/main/scala/mesosphere/marathon/MarathonConf.scala

```diff
@@ -307,5 +307,12 @@ trait MarathonConf
     noshort = true,
     hidden = true,
     default = Some(MesosHeartbeatMonitor.DEFAULT_HEARTBEAT_FAILURE_THRESHOLD))
+
+  lazy val drainingTime = opt[Long](
+    "draining_time",
```

Comment:
Hey @mattmb, thanks for the information! We are more or less following the same approach as you (we have an application that "restarts" the services that has one or more containers in a node in maintenance, forcing a rolling deployment following upgradeStrategy). 

So I understand that the approach that the PR follows would work also for you!